core/parser.mly

```diff
@@ -1068,15 +1068,27 @@ vfield:
 
 efields:
 | fields_def(efield, row_var, kinded_row_var)                  { $1 }
+| DOT                                                          { ( [] , Datatype.DotClosed) }
```

Comment:
I don't think this is correct. The `.` should only be allowed to appear in a row variable position, e.g. the productions `row_var` and `kinded_row_var` should allow `.` and interpret it as `Datatype.Closed`.

Reply:
How about `(a) {.}-> b`, i.e. a function that admits no effects at all?

```links
sig f : (Int, Int) {.}-> Int
fun f(x, y) { x + y }
```
Arguably this is less useful, but if it can happen (is a legal type), I think it should be supported.

We can have this type without the `{.}` syntax, but then it has to be `{ |.}`.

Reply:
`{.}` syntax removed